### PR TITLE
Implement pushdown for `aggregate`

### LINKED ISSUE #3520: Implement pushdown for aggregation pipelines for the new PostgreSQL backend
### What should be done?

Figure out why tests fail and provide the necessary fixes.

The `MsgAggregate()` should support applying filter and sort pushdown in the similar fashion to the `pg/msg_aggregate.go` counterpart:

https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/pg/msg_aggregate.go#L247-L267

### Where?

```
task test-integration-postgresql SHARD_INDEX=0 SHARD_TOTAL=0 TEST_RUN='TestAggregateCompatStages'
task test-integration-postgresql SHARD_INDEX=0 SHARD_TOTAL=0 TEST_RUN='TestAggregateCompatLimit'
task test-integration-postgresql SHARD_INDEX=0 SHARD_TOTAL=0 TEST_RUN='TestAggregateCompatMatch
task test-integration-postgresql SHARD_INDEX=0 SHARD_TOTAL=0 TEST_RUN='TestAggregateCompatSkip'
```

See also this issue mentioned in the code - https://github.com/FerretDB/FerretDB/pull/3522/files.

### Definition of Done

- fixes provided;
- integration/compatibility tests updated.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #3520.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/sqlite/msg_explain.go [hunk 1]
```diff
@@ -78,13 +79,18 @@ func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
-	if !h.DisableFilterPushdown {
-		qp.Filter = params.Filter
+	qp := backends.ExplainParams{
+		Filter: params.Filter,
+	}
+
+	sort := params.Sort
+
+	if params.Aggregate {
+		qp.Filter, sort = aggregations.GetPushdownQuery(params.StagesDocs)
```
Comment (chilagrow) [inline internal/handlers/sqlite/msg_explain.go:89]: `sort` is not pushed down for aggregation but `params.Sort` is pushed down as far as I can see. Correct me if I'm wrong, `params.Sort` is nil for aggregation right? Could you add a test for it please and run it with `EnableSortPushdown` true please :hugs: 

```suggestion
		qp.Filter, params.Sort = aggregations.GetPushdownQuery(params.StagesDocs)
```

**Review by chilagrow — CHANGES_REQUESTED** [review]
I think logic is not correct

File: internal/backends/postgresql/collection.go [hunk 2]
```diff
@@ -314,6 +314,8 @@ func (c *collection) Explain(ctx context.Context, params *backends.ExplainParams
 
 	res.QueryPushdown = where != ""
 
+	q += where
```
Comment (chilagrow) [inline internal/backends/postgresql/collection.go:317]: If sort pushdown was enabled, explain didn't work before :sweat_smile:

**Review by chilagrow — APPROVED** [review]
Thanks for fixing the bug as well.

File: internal/backends/postgresql/collection.go (hunk 2 above)
Comment (noisersup) [reply internal/backends/postgresql/collection.go:318]: Nice catch! Note for myself: our tests for `EnableSortPushdown` **work** :)

**Review by AlekSi — APPROVED** [review]

**Pull request merged**